cli/src/main.rs

```diff
@@ -273,14 +273,26 @@ fn create_settings(args: &Args) -> anyhow::Result<FormatterSettings> {
                 .with_context(|| format!("failed to load config file: {}", path.display()))
         })
         .unwrap_or_else(|| {
-            let default_config: PathBuf = "leptosfmt.toml".into();
```

Comment:
how about extracting this to a separate function `find_config_file`? I think we can use `ancestors` to use the iterator API instead of an imperative loop (I prefer functional style, but this is more preference than anything else):

```rust
fn find_config_file() -> anyhow::Result<Option<PathBuf>> {
    Ok(fs::canonicalize(env::current_dir()?)?
        .ancestors()
        .map(|p| p.join("leptosfmt.toml"))
        .find(|p| p.exists()))
}
```

Reply:
then we can combine everything by using `transpose`:
```rust
    let mut settings = args
        .config_file
        .as_ref()
        .map(load_config)
        .transpose()
        .or_else(|_| find_config_file()?.as_ref().map(load_config).transpose())?
        .unwrap_or_default();
```

Reply:
Updated, with some changes from your suggestion - `.or_else` in the suggestion would run if there was an error loading the config, we actually want to search only if a config file wasn't found, but fail on error.

Reply:
Ah you are correct, we indeed want to fail in that case.